src/site/content/en/blog/nfc/index.md

```diff
@@ -0,0 +1,906 @@
+      },
+      {
+        recordType: "mime", // another icon record
+        mediaType: "image/jpg",
+        data: await (await fetch("icon2.jpg")).arrayBuffer()
+      }
+    ]
+  }
+};
+
+const writer = new NDEFWriter();
+await writer.write({ records: [smartPosterRecord] });
+```
+
+### Read and write an external type record
+
+To create application defined records, use external type records that may
+contain an NDEF message as payload that is accessible with `toRecords()`. Their
+name contains the domain name of the issuing organization, a colon and a type
+name that is at least one character long, for instance "example.com:foo".
+
+```js
+function readExternalTypeRecord(externalTypeRecord) {
+  for (const record of externalTypeRecord.toRecords()) {
+    if (record.recordType == "text") {
+      const decoder = new TextDecoder(record.encoding);
+      console.log(`Text: ${textDecoder.decode(record.data)} (${record.lang})`);
+    } else if (record.recordType == "url") {
+      const decoder = new TextDecoder();
+      console.log(`URL: ${decoder.decode(record.data)}`);
+    } else {
+      // TODO: Handle other type of records.
+    }
+  }
+}
+```
+
+To write an external type record, pass an NDEF message dictionary to the
+NDEFWriter `write()` method. The external type record contained in the NDEF
+message is defined as an object with a `recordType` key set to the name of the
+external type and a `data` key set to an object that represents (once again) an
+NDEF message contained in the external type record.  Note that the `data` key
+can also be either an ArrayBuffer or provides a view on to an ArrayBuffer (.e.g.
+Uint8Array, DataView).
+
+```js
+const externalTypeRecord = {
+  recordType: "example.game:a",
+  data: {
+    records: [
+      {
+        recordType: "url",
+        data: "https://example.game/42"
+      },
+      {
+        recordType: "text",
+        data: "Game context given here"
+      },
+      {
+        recordType: "mime",
+        mediaType: "image/png",
+        data: await (await fetch("image.png")).arrayBuffer()
+      }
+    ]
+  }
+};
+
+const writer = new NDEFWriter();
+writer.write({ records: [externalTypeRecord] });
+```
+
+### Read and write an empty record
+
+An empty record has no payload.
+
+To write an empty record, pass an NDEF message dictionary to the NDEFWriter
+`write()` method. The empty record contained in the NDEF message is defined as
+an object with a `recordType` key set to `"empty"`.
+
+```js
+const emptyRecord = {
+  recordType: "empty"
+};
+
+const writer = new NDEFWriter();
+await writer.write({ records: [emptyRecord] });
+```
+
+## Dev Tips
+
+Here's a list of things I wish I had known when I started playing with Web NFC:
+
+- Android natively handles NFC tags before Web NFC is operational.
+- An NFC icon can be found on [material.io].
+- Use NDEF record `id` to simply identifying a record when needed.
+- An unformatted NFC tag that supports NDEF contains an NDEF message with an
```

Comment:
contains only a single record of the empty type

Reply:
Fixed.